folium/plugins/timeline.py

```diff
@@ -0,0 +1,313 @@
+    ...                    hash = str.charCodeAt(i) + ((hash << 5) - hash);
+    ...                }
+    ...                var red = (hash >> 24) & 0xff;
+    ...                var grn = (hash >> 16) & 0xff;
+    ...                var blu = (hash >> 8) & 0xff;
+    ...                return "rgb(" + red + "," + grn + "," + blu + ")";
+    ...            }
+    ...            return {
+    ...                stroke: false,
+    ...                color: getColorFor(data.properties.name),
+    ...                fillOpacity: 0.5,
+    ...            };
+    ...        }
+    ...    '''
+    ...     ),
+    ... ).add_to(m)
+    >>> GeoJsonPopup(fields=["name"], labels=True).add_to(timeline)
+    >>> TimelineSlider(
+    ...     auto_play=False,
+    ...     enable_keyboard_controls=True,
+    ...     playback_duration=60000,
+    ... ).add_timelines(timeline).add_to(m)
+
+    Other keyword arguments are passed to the GeoJson layer, so you can pass
+      `style`, `point_to_layer` and/or `on_each_feature`.
+
+    """
+
+    _template = Template(
+        """
+        {% macro script(this, kwargs) %}
+          var {{ this.get_name() }}_options = {{ this.options|tojson }};
+          {% for key, value in this.functions.items() %}
+            {{ this.get_name() }}_options["{{key}}"] = {{ value }};
+          {% endfor %}
+
+          var {{ this.get_name() }} = L.timeline(
+              {{ this.data|tojson }},
+              {{ this.get_name() }}_options
+          );
+          {{ this.get_name() }}.addTo({{ this._parent.get_name() }});
+        {% endmacro %}
+    """
+    )
+
+    default_js = [
+        (
+            "timeline",
+            "https://skeate.dev/Leaflet.timeline/examples/leaflet.timeline.js",
+        ),
+        (
+            "moment",
+            "https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.18.1/moment.min.js",
+        ),
+    ]
+
+    def __init__(
+        self,
+        data: Union[dict, str, TextIO],
+        date_options: str = "YYYY-MM-DD HH:mm:ss",
+        point_to_layer: Optional[JsCode] = None,
+        style: Optional[JsCode] = None,
+        on_each_feature: Optional[JsCode] = None,
+        get_interval: Optional[JsCode] = None,
+        **kwargs
+    ):
+        super().__init__(data)
+        self._name = "Timeline"
+
+        kwargs["point_to_layer"] = point_to_layer
+        kwargs["on_each_feature"] = on_each_feature
+        kwargs["style"] = style
+
+        if get_interval is not None:
+            kwargs["get_interval"] = get_interval
+
+        # extract JsCode objects
+        self.functions = {}
+        for key, value in list(kwargs.items()):
+            if isinstance(value, JsCode):
+                self.functions[camelize(key)] = value.js_code
+                kwargs.pop(key)
+
+        self.options = parse_options(**kwargs)
+
+    def render(self, **kwargs):
+        assert isinstance(
+            self._parent, Map
+        ), "Timeline can only be added to a Map object."
```

Comment:
is that true? So it can't be grouped in a featuregroup?

Reply:
Changed.